go/vt/vtgate/planbuilder/testdata/aggr_cases.json

```diff
@@ -5312,22 +5277,28 @@
         ],
         "Inputs": [
           {
-            "OperatorType": "Aggregate",
-            "Variant": "Ordered",
-            "Aggregates": "max(1) AS bazo",
-            "GroupBy": "(0|2)",
+            "OperatorType": "Filter",
+            "Predicate": "bazo between 100 and 200",
             "Inputs": [
               {
-                "OperatorType": "Route",
-                "Variant": "Scatter",
-                "Keyspace": {
-                  "Name": "user",
-                  "Sharded": true
-                },
-                "FieldQuery": "select foo, max(baz) as bazo, weight_string(foo) from (select foo, baz from `user` where 1 != 1) as f where 1 != 1 group by foo, weight_string(foo)",
-                "OrderBy": "(0|2) ASC",
-                "Query": "select foo, max(baz) as bazo, weight_string(foo) from (select foo, baz from `user` having max(baz) between 100 and 200) as f group by foo, weight_string(foo) order by foo asc",
-                "Table": "`user`"
```

Comment:
this looks more optimized to me.
filter in the form of `having` done at the route.

Reply:
probably can be done for max, min only

Reply:
maybe this is wrong, on a single shard it may be true but when done across the shard on a grouping it might not be a valid row on the filter.